sssd_test_framework/roles/keycloak.py

```diff
@@ -0,0 +1,299 @@
+        self.id = result.stdout.split()[-1].strip("'")
+        return self
+
+    def add_member(self, member: KeycloakUser | KeycloakGroup) -> KeycloakGroup:
+        """
+        Add group member.
+
+        :param member: User or group to add as a member.
+        :type member: IPAUser | IPAGroup
+        :return: Self.
+        :rtype: IPAGroup
+        """
+        return self.add_members([member])
+
+    def add_members(self, members: list[KeycloakUser | KeycloakGroup]) -> KeycloakGroup:
+        """
+        Add multiple group members.
+
+        :param member: List of users or groups to add as members.
+        :type member: list[IPAUser | IPAGroup]
+        :return: Self.
+        :rtype: IPAGroup
+        """
+        # run command to add
+        for item in self.__get_member_args(members):
+            update_group = f"update {item}/groups/{self.id} -r master " \
+                        f""
+        return self
+
+    def remove_member(self, member: KeycloakUser | KeycloakGroup) -> KeycloakGroup:
+        """
+        Remove group member.
+
+        :param member: User or group to remove from the group.
+        :type member: IPAUser | IPAGroup
+        :return: Self.
+        :rtype: IPAGroup
+        """
+        return self.remove_members([member])
+
+    def remove_members(self, members: list[KeycloakUser | KeycloakGroup]) -> KeycloakGroup:
+        """
+        Remove multiple group members.
+
+        :param member: List of users or groups to remove from the group.
+        :type member: list[KeycloakUser | KeycloakGroup]
+        :return: Self.
+        :rtype: KeycloakGroup
+        """
+        self._exec('remove-member', self.__get_member_args(members))
+        return self
+
+    def __get_member_args(self, members: list[KeycloakUser | KeycloakGroup]) -> list[str]:
+        users = [x for item in members if isinstance(item, KeycloakUser) for x in (f'users/{item.id}',)]
+        groups = [x for item in members if isinstance(item, KeycloakGroup) for x in (f'groups/{item.id}',)]
+        return [*users, *groups]
+
+
+class KeycloakDomain(object):
+    """
+    Keycloak integration domain management.
+    """
+
+    def __init__(self, role: Keycloak, name: str) -> None:
+        """
+        :param role: Keycloak role object.
+        :type role: Keycloak
+        :param name: Integration Domain name.
+        :type name: str
+        """
+        self.role = role
+        self.name = name
+        self.id: str = ""
+
+    def add(
+        self,
+        *,
+        scim_host: str,
+        scim_admin: str | None = "scim",
+        scim_password: str | None = "Secret123",
+        dom_name: str,
+        dom_url: str,
+        client_id: str | None = "admin",
+        client_secret: str | None = "Secret123",
+        id_provider: str | None = "ipa",
+        ldap_tls_cacert: str | None = None,
+    ) -> KeycloakDomain:
+        """
+        Create new Keycloak integration domain.
+
```

Comment:
Please add docstrings for each parameter.

Reply:
adding